tests/lib/test_openstack_flavor.py

```diff
@@ -0,0 +1,211 @@
+import unittest
+from unittest.mock import NonCallableMock, Mock, MagicMock, call
+from openstack_api.openstack_flavor import OpenstackFlavor
+
+
+# pylint:disable=too-many-public-methods
+class OpenstackFlavorTests(unittest.TestCase):
+    """
+    Runs various tests to ensure we are using the Openstack
+    Flavor module in the expected way
+    """
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.mocked_connection = MagicMock()
+        self.instance = OpenstackFlavor(self.mocked_connection)
+        self.api = self.mocked_connection.return_value.__enter__.return_value
+
+    def test_list_flavor(self):
+        """
+        Test that we can retrieve a list of flavors from a specific cloud
+        """
+        mocked_name = NonCallableMock()
+        flavors = self.instance.list_flavor(mocked_name)
+        assert flavors == self.api.list_flavors.return_value
+        self.mocked_connection.assert_called_once_with(mocked_name)
+
+    def test_get_flavor(self):
+        """
+        Test that we can get information for a specific flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        mocked_flavor_name = NonCallableMock()
+        flavor = self.instance.get_flavor(mocked_cloud_name, mocked_flavor_name)
+
+        assert flavor == self.api.get_flavor.return_value
+
+        self.api.get_flavor.assert_called_once_with(mocked_flavor_name)
+        self.mocked_connection.assert_called_once_with(mocked_cloud_name)
+
+    def test_get_missing_flavors_calls_list_flavor(self):
+        """
+        Test that we are passing the source and destination clouds in the correct order
+        in order to list flavors from each cloud
+        """
+
+        source_cloud, dest_cloud = NonCallableMock(), NonCallableMock()
+        self.instance.list_flavor = Mock(return_value=[])
+        self.instance.get_missing_flavors(source_cloud, dest_cloud)
+        self.instance.list_flavor.assert_has_calls(
+            [call(source_cloud), call(dest_cloud)],
+            any_order=False,  # Order matters here
+        )
+
+    def test_get_missing_flavors_get_diff(self):
+        """
+        Test getting the difference between list of flavor names
+        """
+        mock_source_names = [NonCallableMock() for _ in range(3)]
+        for i, obj in enumerate(mock_source_names):
+            obj.name = i
+
+        mock_dest_names = [NonCallableMock() for _ in range(3)]
+        for i, obj in enumerate(mock_dest_names):
+            obj.name = i + 2
+
+        mock_side_effect = [mock_source_names, mock_dest_names]
+
+        self.instance.list_flavor = Mock(side_effect=mock_side_effect)
+        return_diff = self.instance.get_missing_flavors(
+            NonCallableMock(), NonCallableMock()
+        )
+        assert [0, 1] == return_diff
```

Comment:
A small comment here would help visualise why 0, 1. Something alone the lines of 
```
# We expect 0, 1 since the first set contains 0:3 and the second set contains 2:5
# so 0, 1 are the elements missing in set two
```

Feel free to reword this